packages/cli/src/lib/commands/locations/modes-util.ts

```diff
@@ -0,0 +1,54 @@
+import { Errors } from '@oclif/core'
+
+import { LocationItem, Mode, SmartThingsClient } from '@smartthings/core-sdk'
+
+import * as modesUtil from './modes-util'
+import { APICommand, selectFromList, SelectFromListConfig, TableFieldDefinition, WithNamedLocation } from '@smartthings/cli-lib'
+
+
+export const tableFieldDefinitions: TableFieldDefinition<Mode>[] = ['label', 'id', 'name' ]
+export const tableFieldDefinitionsWithLocationName: TableFieldDefinition<Mode & WithNamedLocation>[] = ['label', 'id', 'name', 'location', 'locationId' ]
+
+export async function getModesByLocation(client: SmartThingsClient, locationId?: string): Promise<(Mode & WithNamedLocation)[]> {
+	let locations: LocationItem[] = []
+	if (locationId) {
+		locations = [await client.locations.get(locationId)]
+	} else {
+		locations = await client.locations.list()
+	}
```

Comment:
Nothing at all wrong with this code but you could use a ternary to avoid the `let`:

```
	const locations = locationId
		? [await client.locations.get(locationId)]
		: await client.locations.list()
```

Reply:
Much better, thanks!